tm2/pkg/bft/abci/types/types.go

```diff
@@ -116,6 +117,32 @@ func (r ResponseBase) IsErr() bool {
 	return r.Error != nil
 }
 
+func (r ResponseBase) EventString() string {
+	levts := len(r.Events)
+	if levts == 0 {
+		return `""`
+	}
+
+	var builder strings.Builder
+
+	if levts > 1 {
+		builder.WriteString("[")
+	}
+
+	for i, event := range r.Events {
+		if i > 0 {
+			builder.WriteString(",")
+		}
+		builder.WriteString(event.String())
+	}
+
+	if levts > 1 {
+		builder.WriteString("]")
+	}
+
+	return builder.String()
+}
```

Comment:
nitpick: avoid double len check and overlapping logic.

```suggestion
func (r ResponseBase) EventString() string {
	switch len(r.Events) {
	case 0:
		return `""`
	case 1:
		return event.String()
	default: // > 1
	}

	var builder strings.Builder

	builder.WriteRune('[')
	for i, event := range r.Events {
		if i > 0 {
			builder.WriteRune(',')
		}
		builder.WriteString(event.String())
	}
	builder.WriteRune(']')

	return builder.String()
}
```